xarray/backends/zarr.py

```diff
@@ -679,8 +685,17 @@ def set_variables(self, variables, check_encoding_set, writer, unlimited_dims=No
 
                 if coding.strings.check_vlen_dtype(dtype) == str:
                     dtype = str
+
+                # If user has specified write_empty_chunks through per-variable encoding, prefer that.
+                if "write_empty_chunks" not in encoding:
+                    encoding["write_empty_chunks"] = self._write_empty
```

Comment:
Does encoding normally take priority over user input? 

I'm thinking if you change the default to None instead, `write_empty_chunks: bool | None = None`, it'll be possible to override encoding.

Reply:
I took both approaches, using a default value of `None` where it will fall back to encoding / Zarr defaults. And an error is raised if encoding and the parameter differ.